src/mudlet.cpp

```diff
@@ -2417,3 +2423,73 @@ void mudlet::requestProfilesToReloadMaps( QList<QString> affectedProfiles )
 {
     emit signal_profileMapReloadRequested( affectedProfiles );
 }
+
+void mudlet::slot_gamepadButtonPress(int deviceId, QGamepadManager::GamepadButton button, double value)
+{
+    Host * pH = getActiveHost();
+    if( ! pH ) return;
+    TEvent event;
+    event.mArgumentList.append( "sysGamepadButtonPress" );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+    event.mArgumentList.append( QString::number(deviceId) );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    event.mArgumentList.append( QString::number(button) );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    event.mArgumentList.append( QString::number(value) );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    pH->raiseEvent( event );
+}
+
+void mudlet::slot_gamepadButtonRelease(int deviceId, QGamepadManager::GamepadButton button)
+{
+    Host * pH = getActiveHost();
+    if( ! pH ) return;
+    TEvent event;
+    event.mArgumentList.append( "sysGamepadButtonRelease" );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+    event.mArgumentList.append( QString::number(deviceId) );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    event.mArgumentList.append( QString::number(button) );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    pH->raiseEvent( event );
+}
+
+void mudlet::slot_gamepadConnected(int deviceId)
+{
+    Host * pH = getActiveHost();
+    if( ! pH ) return;
+    TEvent event;
+    event.mArgumentList.append( "sysGamepadConnected" );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_STRING);
+    event.mArgumentList.append( QString::number(deviceId) );
+    event.mArgumentTypeList.append(ARGUMENT_TYPE_NUMBER);
+    pH->raiseEvent( event );
```

Comment:
Is there any further information about the connected device (like manufacturer/product identification strings) that can help to identify/differentiate between multiple devices?